tools/tpm2_createpolicy.c

```diff
@@ -0,0 +1,358 @@
+    if (rval != TPM_RC_SUCCESS) {
+        return rval;
+    }
+
+    rval = Tss2_Sys_PolicyPCR(ctx->sapi_context,
+            ctx->policy_session->sessionHandle, 0, &pcr_digest, &pcrs, 0);
+
+    return rval;
+}
+
+TPM_RC build_policy(createpolicypcr_ctx *ctx,
+        TPM_RC (*build_policy_function)(createpolicypcr_ctx *ctx)) {
+    // NOTE:  this policy_session will be a trial policy session
+    TPM2B_ENCRYPTED_SECRET encryptedSalt = { { 0 }, };
+    TPMT_SYM_DEF symmetric = { .algorithm = TPM_ALG_NULL, };
+
+    TPM2B_NONCE nonceCaller = { { 0, } };
+    nonceCaller.t.size = 0;
+
+    // Start policy session.
+    TPM_RC rval = tpm_session_start_auth_with_params(ctx->sapi_context,
+            &ctx->policy_session, TPM_RH_NULL, 0, TPM_RH_NULL, 0, &nonceCaller,
+            &encryptedSalt, TPM_SE_TRIAL, &symmetric, TPM_ALG_SHA256);
+
+    if (rval != TPM_RC_SUCCESS) {
+        LOG_ERR("Failed tpm session start auth with params\n");
+        return rval;
+    }
+
+    // Send policy command.
+    rval = (*build_policy_function)(ctx);
+    if (rval != TPM_RC_SUCCESS) {
+        LOG_ERR("Failed build_policy_function\n");
+        return rval;
+    }
+
+    // Get policy hash.
+    ctx->policy_digest.t.size = 0;
+    INIT_SIMPLE_TPM2B_SIZE(ctx->policy_digest);
+    rval = Tss2_Sys_PolicyGetDigest(ctx->sapi_context,
+            ctx->policy_session->sessionHandle, 0, &ctx->policy_digest, 0);
+    if (rval != TPM_RC_SUCCESS) {
+        LOG_ERR("Failed Policy Get Digest\n");
+        return rval;
+    }
+
+    LOG_INFO("Created Policy Digest:\n");
+
+    //save the policy buffer in a file for use later
+    bool result = files_save_bytes_to_file(ctx->policyfile, (UINT8 *) &ctx->policy_digest.t.buffer,
+            ctx->policy_digest.t.size);
+    if (!result) {
+        LOG_ERR("Failed to save policy digest into file \"%s\"",
+                ctx->policyfile);
+        return -1;
+    }
+
+    // Need to flush the session here.
+    rval = Tss2_Sys_FlushContext(ctx->sapi_context,
+            ctx->policy_session->sessionHandle);
+    if (rval != TPM_RC_SUCCESS) {
+        LOG_ERR("Failed Flush Context\n");
+        return rval;
+    }
+
+    // And remove the session from sessions table.
+    return tpm_session_auth_end(ctx->policy_session);
+}
+
+bool parse_policy_type(createpolicypcr_ctx *ctx, char *argv[]) {
+
+    if (ctx->pcr_flags.policy_type_pcr_flag) {
+        if (!ctx->policy_file_flag || !ctx->pcr_flags.pcr_index_flag) {
+            showArgMismatch(argv[0]);
+            return false;
+        } else {
+            LOG_INFO("Policy File = %s\n", ctx->policyfile);
+            LOG_INFO("PCR Index= %d\n", ctx->pcr_index);
+            UINT32 rval = build_policy(ctx, build_pcr_policy);
+            if (rval != TPM_RC_SUCCESS) {
+                LOG_ERR("Failed build_policy\n");
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+#define MAX_SUPPORTED_PCRS 24
```

Comment:
As in should we have a routine to read this out from the tpm capability structure?
Also is it fixed in TSS. I guess yes.